pkg/activator/throttler_test.go

```diff
@@ -0,0 +1,249 @@
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	revID = RevisionID{"good-namespace", "good-name"}
+
+	sampleError = "some error"
+
+	existingRevisionGetter = func(concurrency v1alpha1.RevisionContainerConcurrencyType) func(RevisionID) (*v1alpha12.Revision, bool, error) {
+		return func(RevisionID) (*v1alpha12.Revision, bool, error) {
+			revision := &v1alpha12.Revision{Spec: v1alpha12.RevisionSpec{ContainerConcurrency: concurrency}}
+			return revision, true, nil
+		}
+	}
+	nonExistingRevisionGetter = func(RevisionID) (*v1alpha12.Revision, bool, error) {
+		revision := &v1alpha12.Revision{}
+		return revision, false, nil
+	}
+	initCapacity            = int32(0)
+	existingEndpointsGetter = func(RevisionID) (int32, error) {
+		return initCapacity, nil
+	}
+	nonExistingEndpointsGetter = func(RevisionID) (int32, error) {
+		return initCapacity, errors.New(sampleError)
+	}
+)
+
+const (
+	defaultMaxConcurrency = int32(10)
+)
+
+func TestThrottler_UpdateCapacity(t *testing.T) {
+	samples := []struct {
+		label           string
+		revisionGetter  func(RevisionID) (*v1alpha12.Revision, bool, error)
+		endpointsGetter func(RevisionID) (int32, error)
+		maxConcurrency  int32
+		want            int32
+		wantError       string
+	}{
+		{
+			label:           "all good",
+			revisionGetter:  existingRevisionGetter(10),
+			endpointsGetter: existingEndpointsGetter,
+			maxConcurrency:  defaultMaxConcurrency,
+			want:            int32(10),
+		},
+		{
+			label:           "unlimited concurrency",
+			revisionGetter:  existingRevisionGetter(0),
+			endpointsGetter: existingEndpointsGetter,
+			maxConcurrency:  100,
+			want:            int32(100),
+		},
+		{
+			label:           "non-existing revision",
+			revisionGetter:  nonExistingRevisionGetter,
+			endpointsGetter: existingEndpointsGetter,
+			maxConcurrency:  defaultMaxConcurrency,
+			want:            int32(0),
+			wantError:       fmt.Sprintf("no revision was found for: %s", revID),
+		},
+	}
+	for _, s := range samples {
+		t.Run(s.label, func(t *testing.T) {
+			var received string
+			want := s.want
+			throttler := getThrottler(s.maxConcurrency, s.revisionGetter, s.endpointsGetter, t)
+			err := throttler.UpdateCapacity(revID, 1)
+			if s.wantError != "" {
+				received = err.Error()
+			}
+			if received != s.wantError {
+				t.Errorf("Expected error in Update capacity. Want %s, got %s", s.wantError, err.Error())
+			}
+			if want > 0 {
+				breaker, _ := throttler.breakers[revID]
+				got := breaker.Capacity()
+				if got != want {
+					t.Errorf("Unexpected capacity of the breaker. Want %d, got %d", want, got)
+				}
+			}
+		})
+	}
+}
+
+func TestThrottler_Try(t *testing.T) {
+	samples := []struct {
+		label           string
+		addCapacity     bool
+		wantBreakers    int32
+		wantError       string
+		revisionGetter  func(RevisionID) (*v1alpha12.Revision, bool, error)
+		endpointsGetter func(RevisionID) (int32, error)
+	}{
+		{
```

Comment:
You can collapse these cases vertically by using `{{` here and `}, {` between cases. `}}` at the end.